### PR TITLE
[Merged by Bors] - chore(Algebra): improve argument names to induction principles

### PR DESCRIPTION
**Pull request by eric-wieser**
These are the case names used by the `induction` tactic after the `with`.

This replaces `H0`, `H1`, `Hmul` etc with `zero`, `one`, `mul`.

This PR does not touch `Submonoid` or `Subgroup`, as `to_additive` does not know how to rename the argument names.
There are ways to work around this, but I'd prefer to leave them to a later PR.

This also leaves the `closure_induction₂` variants alone, as renaming the arguments is more work for less gain.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Star/Subalgebra.lean [hunk 1]
```diff
@@ -497,12 +497,12 @@ theorem _root_.Subalgebra.starClosure_eq_adjoin (S : Subalgebra R A) :
 /-- If some predicate holds for all `x ∈ (s : Set A)` and this predicate is closed under the
 `algebraMap`, addition, multiplication and star operations, then it holds for `a ∈ adjoin R s`. -/
 theorem adjoin_induction {s : Set A} {p : A → Prop} {a : A} (h : a ∈ adjoin R s)
-    (Hs : ∀ x : A, x ∈ s → p x) (Halg : ∀ r : R, p (algebraMap R A r))
-    (Hadd : ∀ x y : A, p x → p y → p (x + y)) (Hmul : ∀ x y : A, p x → p y → p (x * y))
-    (Hstar : ∀ x : A, p x → p (star x)) : p a :=
+    (mem : ∀ x : A, x ∈ s → p x) (algebraMap : ∀ r : R, p (algebraMap R A r))
+    (add : ∀ x y : A, p x → p y → p (x + y)) (Hmul : ∀ x y : A, p x → p y → p (x * y))
```
Comment (j-loreaux) [inline Mathlib/Algebra/Star/Subalgebra.lean:501]:
```suggestion
    (add : ∀ x y : A, p x → p y → p (x + y)) (mul : ∀ x y : A, p x → p y → p (x * y))
```

File: Mathlib/FieldTheory/Subfield.lean [hunk 2]
```diff
@@ -686,14 +686,14 @@ theorem closure_eq_of_le {s : Set K} {t : Subfield K} (h₁ : s ⊆ t) (h₂ : t
 of `s`, and is preserved under addition, negation, and multiplication, then `p` holds for all
 elements of the closure of `s`. -/
 @[elab_as_elim]
-theorem closure_induction {s : Set K} {p : K → Prop} {x} (h : x ∈ closure s) (Hs : ∀ x ∈ s, p x)
-    (H1 : p 1) (Hadd : ∀ x y, p x → p y → p (x + y)) (Hneg : ∀ x, p x → p (-x))
-    (Hinv : ∀ x, p x → p x⁻¹) (Hmul : ∀ x y, p x → p y → p (x * y)) : p x := by
+theorem closure_induction {s : Set K} {p : K → Prop} {x} (h : x ∈ closure s) (mem : ∀ x ∈ s, p x)
+    (one : p 1) (add : ∀ x y, p x → p y → p (x + y)) (neg : ∀ x, p x → p (-x))
+    (Hinv : ∀ x, p x → p x⁻¹) (mul : ∀ x y, p x → p y → p (x * y)) : p x := by
```
Comment (j-loreaux) [inline Mathlib/FieldTheory/Subfield.lean:691]:
```suggestion
    (inv : ∀ x, p x → p x⁻¹) (mul : ∀ x y, p x → p y → p (x * y)) : p x := by
```

**Review by j-loreaux — COMMENTED** [review]
A few were missed, marked below. Otherwise it looks good!

**Comment by j-loreaux** [comment]
bors d+

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**